packages/snowflake/src/lib/Snowflake.ts

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable @typescript-eslint/explicit-member-accessibility */
+
+/**
+ * A class for parsing snowflake ids
+ */
+export class Snowflake {
+	/**
+	 * Internal incrementer for generating snowflakes
+	 * @internal
+	 */
+	#increment = 0n;
+
+	/**
+	 * Internal reference of the epoch passed in the constructor
+	 * @internal
+	 */
+	#epoch: bigint;
+
+	/**
+	 * Alias for {@link deconstruct}
+	 */
+	// eslint-disable-next-line @typescript-eslint/unbound-method, @typescript-eslint/no-invalid-this
+	public decode = this.deconstruct;
+
+	/**
+	 * @param epoch the epoch to use
+	 */
+	public constructor(epoch: bigint) {
```

Comment:
There's a mismatch between the readme and the types here... You allow people to pass in a Date object, or possibly a number, while taking just a bigint per the types.. can you handle those edge cases? In the end, the epoch should be kept as a bigint internally, but the constructor should handle it